eng/common/scripts/check-spelling-in-changed-files.ps1

```diff
@@ -144,10 +39,311 @@ Optional location to use for cspell.json path. Default value is
 .PARAMETER ExitWithError
 Exit with error code 1 if spelling errors are detected.
 
+.PARAMETER Test
+Run test functions against the script logic
+
 .EXAMPLE
 ./eng/common/scripts/check-spelling-in-changed-files.ps1 -TargetBranch 'target_branch_name'
 
 This will run spell check with changes in the current branch with respect to 
 `target_branch_name`
 
 #>
+
+[CmdletBinding()]
+Param (
+    [Parameter()]
+    [string] $TargetBranch,
+
+    [Parameter()]
+    [string] $SourceBranch,
+
+    [Parameter()]
+    [string] $CspellConfigPath = "./.vscode/cspell.json",
+
+    [Parameter()]
+    [switch] $ExitWithError,
+
+    [Parameter()]
+    [switch] $Test
+)
+
+function CheckSpellingInChangedFilesImpl() {
+    $ErrorActionPreference = "Continue"
```

Comment:
Should these 2 lines be moved to the outer scope? Also should we be importing the common.ps1 in this context?